test/tests/api/v1_1/nodes_tests.py

```diff
@@ -106,9 +106,11 @@ def get_undiscovered_compute_count(self):
                 workflows = self.__get_data()
                 LOG.info(workflows)
                 for workflow in workflows:
-                    if workflow.get['injectableName'] == 'Graph.SKU.Discovery' \
-                     and workflow.get['status'] == 'succeeded':
-                     discovered_count += 1
+                    definition = workflow.get('definition', {})
+                    injectableName = definition.get('injectableName')
+                    if injectableName == 'Graph.SKU.Discovery':
+                        workflow.get['_status'] == 'succeeded':
```

Comment:
yeah i see "You're assuming the status will be correct which we have seen in the past its not always the case, The status in AMQP is probably more reliable" But why it's not reliable